joomla/joomla-cms
# Update sites improvements

This fixes https://github.com/joomla/joomla-cms/pull/5284
## Description

This pull request tries to fix three problems:
### 1. Disabling update sites

Joomla disables update sites when connecting to them fails. It may be caused by update sites not being available temporary. So disabling them is the worse we can do because it leaves the sites disconnected from updates.

This pull request removes the automatic disabling way. If there is a connection error to the update sites they will be still shown when searching for updates. 

Users still can disable the update sites manually.
### 2. Already disabled update sites

There are sites that already have some update sites disabled and users don't know it. This pull request adds a warning when you click on `Find updates` on `Extension Manager: Update` if there are existing disabled update sites:

![warnin-extension-update](https://cloud.githubusercontent.com/assets/1119272/5279405/a70f4154-7ae4-11e4-9a28-68722a53f2bb.png)
### 3. Language updates

When users go to the `Language Installer` they may not know that the languages update site has been disabled. When user clicks on `Find Languages` we can be sure that they want to enable the update site because they are searching for new languages. So this pull request automatically enables the languages update site when clicking in `Find Languages`.
## Backward compatibility

This pull request should not cause any B/C issue
## Test

To test after applying the patch:
- Edit your database and set `enabled` to 0 in the `#__update_sites` table for `Accredited Joomla! Translations`. Then go to the language manager and click on `Find Languages`. The site should be enabled automatically and language updates found.
- Edit your database and modify the location of any entry in the `#__update_sites` to make sure it's wrong. Then go to `Extension Manager: Update` and make sure that when clicking on `Find Updates` that entry is not disabled but a warning about the wrong connection is still shown. You can check that before the patch it was disabled.
- Edit your database and set `enabled` to 0 in the `#__update_sites` for any entry. Then go to `Extension Manager: Update` and click in `Find Updates`. You should see the disabled update sites warning and the link should point to the `Update Sites Manager`
